rust-runtime/aws-smithy-runtime-api/src/type_erasure.rs

```diff
@@ -224,6 +224,25 @@ impl TypeErasedError {
     }
 }
 
+impl From<Box<dyn StdError + Send + Sync>> for TypeErasedError {
```

Comment:
Needs tests.

Reply:
It's good you suggested this b/c I discovered you can't downcast back to a `BoxError`.

I wrote this test but it doesn't compile. I'll see if I can think of another way to handle this.
```rust
    #[test]
    fn test_typed_erased_errors_can_be_downcast() {
        let test_err: BoxError = TestErr::new("something failed!").into();
        let type_erased_test_err: TypeErasedError = test_err.into();
        let actual = type_erased_test_err.downcast::<Box<dyn std::error::Error + Send + Sync + 'static>>().expect("type erased error can be downcast into original type");
        let expected: BoxError = TestErr::new("something failed!").into();
        assert_eq!(test_err.to_string(), actual.to_string());
    }
```

Reply:
I thought of a better way. Thanks for your advice John.